src/response/chi0.jl

```diff
@@ -146,31 +145,32 @@ function sternheimer_solver(Hk, ψk, εnk, rhs, n;
     # 1     N_occupied  N_occupied + N_extra
 
     # ψk_extra are not converged but have been Rayleigh-Ritzed (they are NOT
-    # eigenvectors of H) so H(ψk_extra) = ψk_extra' (Hk-εn) ψk_extra should be a
+    # eigenvectors of H) so H(ψk_extra) = ψk_extra' (Hk-ε) ψk_extra should be a
     # real diagonal matrix.
-    H(ϕ) = Hk * ϕ - εnk * ϕ
-    ψk_exHψk_ex = Diagonal(real.(εk_extra .- εnk))
+    H(ϕ) = Hk * ϕ - ε * ϕ
+    ψk_exHψk_ex = Diagonal(real.(εk_extra .- ε))
 
     # 1) solve for δψknᴿ
     # ----------------------------
     # writing αkn as a function of δψknᴿ, we get that δψknᴿ
     # solves the system (in Ran(1-P_computed))
     #
-    # R * (H - εn) * (1 - M * (H - εn)) * R * δψknᴿ = R * (1 - M) * b
+    # R * (H - ε) * (1 - M * (H - ε)) * R * δψknᴿ = R * (1 - M) * b
     #
-    # where M = ψk_extra * (ψk_extra' (H-εn) ψk_extra)^{-1} * ψk_extra'
+    # where M = ψk_extra * (ψk_extra' (H-ε) ψk_extra)^{-1} * ψk_extra'
     # is defined above and b is the projection of -rhs onto Ran(Q).
     #
     b = -Q(rhs)
     bb = R(b -  Hψk_extra * (ψk_exHψk_ex \ ψk_extra'b))
     function RAR(ϕ)
         Rϕ = R(ϕ)
-        # Schur complement of (1-P) (H-εn) (1-P)
+        # Schur complement of (1-P) (H-ε) (1-P)
         # with the splitting Ran(1-P) = Ran(P_extra) ⊕ Ran(R)
         R(H(Rϕ) - Hψk_extra * (ψk_exHψk_ex \ Hψk_extra'Rϕ))
     end
     precon = PreconditionerTPA(basis, kpoint)
-    precondprep!(precon, ψk[:, n])
+    # First column of ψk as there is no natural kinetic energy.
+    precondprep!(precon, ψk[:, 1])
```

Comment:
Wouldn't it be more reasonalbe to use the last column where the KE is larger?

Reply:
Not sure, I think both are reasonable... The preconditioner is not super sensitive to that anyway. We could just as well set it to 1, but here at least we have more chance to give something sensible in the "weird" cases (scaled kinetic energy, super compressed system...)